internal/controllers/nmc_reconciler.go

```diff
@@ -113,11 +115,31 @@ func (r *NMCReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 
 		logger := logger.WithValues("module", moduleNameKey)
 
-		if err := r.helper.ProcessModuleSpec(ctrl.LoggerInto(ctx, logger), &nmcObj, &mod, statusMap[moduleNameKey]); err != nil {
+		updateStatus, err := r.helper.ProcessModuleSpec(ctrl.LoggerInto(ctx, logger), &nmcObj, &mod, statusMap[moduleNameKey])
+		if err != nil {
 			errs = append(
 				errs,
 				fmt.Errorf("error processing Module %s: %v", moduleNameKey, err),
 			)
+
+			continue
+		}
+
+		if updateStatus {
```

Comment:
`SyncStatus` will only happen at the next reconciliation. Meanwhile, we will run `GarbageCollectInUseLabels`; is the status is not populated immediately, we may be removing the in-use label just after creating a worker Pod.

Reply:
GarbageCollection will remove in-use label in case the spec and status for module is missing. Since the NMC object is read prior to SyncStatus and is not read again prior to ProcessModuleSpec or ProcessOrphanModuleStatus or GC, then if the  pod was created it means that either Spec or Status for Module were present, and that means that GC won't remove in-use label. I think...

Reply:
> read prior to SyncStatus and is not read again prior to ProcessModuleSpec or ProcessOrphanModuleStatus or GC

But those functions update `nmcObj`. We work with the same object until garbage collection, so by the time we arrive there, we have the up-to-date NMC and can reliably determine which labels need to be gone.

Reply:
I agree that NMC is up-to-date, but i don't see why we need to update the status right after ProcessModuleSpec. Ypu added it in order for GC to work correctly, but it will work correctly even if you don't update the status ( the new code) after ProcessModuleSpec

Reply:
Indeed. First version of the GC was only working with the statuses, not the specs, hence the extra code. Now removed.